rapids-4-spark-tools/src/main/scala/org/apache/spark/sql/rapids/tool/profiling/ApplicationInfo.scala

```diff
@@ -311,4 +543,123 @@ class ApplicationInfo(
+  def stageMetricsAggregationSQL: String = {
+    s"""select $index as appIndex, concat('stage_',s.stageId) as ID,
+       |count(*) as numTasks, max(s.duration) as Duration
+       |$generateAggSQLString
+       |from taskDF_$index t, stageDF_$index s
+       |where t.stageId=s.stageId
+       |group by s.stageId
+       |""".stripMargin
+  }
+
+  // Function to generate a query for job+stage level Task Metrics aggregation
+  def jobAndStageMetricsAggregationSQL: String = {
+    jobMetricsAggregationSQL + " union " + stageMetricsAggregationSQL
+  }
+
+  // Function to generate a query for SQL level Task Metrics aggregation
+  def sqlMetricsAggregationSQL: String = {
+    s"""select $index as appIndex, '$appId' as appID,
+       |sq.sqlID, sq.description,
+       |count(*) as numTasks, max(sq.duration) as Duration,
+       |round(sum(executorCPUTime)/sum(executorRunTime)*100,2) executorCPURatio
+       |$generateAggSQLString
+       |from taskDF_$index t, stageDF_$index s,
+       |jobDF_$index j, sqlDF_$index sq
+       |where t.stageId=s.stageId
+       |and array_contains(j.stageIds, s.stageId)
+       |and sq.sqlID=j.sqlID
+       |group by sq.sqlID,sq.description
+       |""".stripMargin
+  }
+
+  // Function to generate a query for printing SQL metrics(accumulables)
+  def generateSQLAccums: String = {
+    s"""with allaccums as
+       |(
+       |select s.sqlID, p.nodeID, p.nodeName,
+       |s.accumulatorId, s.name, d.value, s.metricType
+       |from sqlMetricsDF_$index s, driverAccumDF_$index d,
+       |planNodeAccumDF_$index p
+       |where s.sqlID = d.sqlID and s.accumulatorId=d.accumulatorId
+       |and s.sqlID=p.sqlID and s.accumulatorId=p.accumulatorId
+       |union
+       |select s.sqlID, p.nodeID, p.nodeName,
+       |s.accumulatorId, s.name, t.value, s.metricType
+       |from jobDF_$index j, sqlDF_$index sq ,
+       |taskStageAccumDF_$index t, sqlMetricsDF_$index s,
+       |planNodeAccumDF_$index p
+       |where array_contains(j.stageIds, t.stageId)
+       |and sq.sqlID=j.sqlID
+       |and s.sqlID = sq.sqlID
+       |and s.accumulatorId=t.accumulatorId
+       |and s.sqlID=p.sqlID and s.accumulatorId=p.accumulatorId
+       |)
+       |select sqlID, nodeID, nodeName,
+       |accumulatorId, name, max(value) as max_value, metricType
+       |from allaccums
+       |group by sqlID, nodeID, nodeName, accumulatorId, name, metricType
+       |order by sqlID, nodeID, nodeName, accumulatorId, name, metricType
+       |""".stripMargin
+  }
+
+  // Function to generate a query for qualification
+  def qualificationSQL: String = {
+    s"""select $index as appIndex, '$appId' as appID,
+       |sq.sqlID, sq.description,
+       |sq.duration, m.executorCPURatio
+       |from sqlDF_$index sq , sqlAggMetricsDF m
+       |where $index = m.appIndex and sq.sqlID = m.sqlID
+       |and sq.sqlID not in (select sqlID from problematicSQLDF_$index)
+       |and sq.duration > 30000
```

Comment:
eventually we need to make configurable

Reply:
Can we make it as follow on?

Reply:
yes

Reply:
actually you can leave this off and we can handle under the qualification changes